### PR TITLE
Added Google Oauth

### PR DESCRIPTION
**Pull request by gabepolk**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -34,6 +34,11 @@ group :doc do
   gem 'sdoc', require: false
 end
 
+gem 'omniauth'
```
Comment (sdevani) [inline Gemfile]: You don't need this if you have the gem below

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,13 +1,8 @@
 class ApplicationController < ActionController::Base
-  # Prevent CSRF attacks by raising an exception.
-  # For APIs, you may want to use :null_session instead.
```
Comment (sdevani) [inline app/controllers/application_controller.rb:3]: Don't delete these comments

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -1,14 +1,22 @@
...
-  <%= javascript_include_tag "application" %>
-</head>
-<body>
-
-<%= yield %>
-
-</body>
+  <head>
+    <title>Google Auth Example App</title>
+    <%= stylesheet_link_tag    "application", media: "all", "data-turbolinks-track" => true %>
```
Comment (sdevani) [inline app/views/layouts/application.html.erb]: Remove turbolinks
Rename title of app to Pied Piper

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -1,14 +1,22 @@
...
+    <%= javascript_include_tag "application", "data-turbolinks-track" => true %>
+    <%= csrf_meta_tags %>
+  </head>
+  <body>
+    <div>
+      <% if current_user %>
```
Comment (sdevani) [inline app/views/layouts/application.html.erb]: Move this code into a navbar partial

File: app/views/sessions/create.html.erb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+<h1>Sessions#create</h1>
```
Comment (sdevani) [inline app/views/sessions/create.html.erb]: Remove unnecessary views

File: config/initializers/omniauth.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+require 'omniauth-google-oauth2'
+
+Rails.application.config.middleware.use OmniAuth::Builder do
+  provider :google_oauth2, "911135299478-su1ivmh1ul0ellg6pkvd76id8va257dm.apps.googleusercontent.com", "qM9behImaTxyPlEW6aJ1Rsep", { access_type: "offline", approval_prompt: "" }
```
Comment (sdevani) [inline config/initializers/omniauth.rb]: Do not ever commit/push API keys. We'll talk with Mike to figure out good system for environment variables

File: config/locales/devise.en.yml [hunk 7]
```diff
@@ -0,0 +1,59 @@
+# Additional translations at https://github.com/plataformatec/devise/wiki/I18n
```
Comment (sdevani) [inline config/locales/devise.en.yml]: We're not using devise.

File: db/migrate/20140731205054_create_users.rb [hunk 8]
```diff
@@ -1,11 +1,12 @@
 class CreateUsers < ActiveRecord::Migration
```
Comment (sdevani) [inline db/migrate/20140731205054_create_users.rb:1]: You should never edit old migrations. Create a new migration that will add/remove fields

File: spec/views/home/show.html.erb_spec.rb [hunk 9]
```diff
@@ -0,0 +1,5 @@
+require 'spec_helper'
```
Comment (sdevani) [inline spec/views/home/show.html.erb_spec.rb]: Delete unnecessary specs. Delete all unnecessary files

File: app/controllers/sessions_controller.rb [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+    redirect_to root_path
+  end
+
+  def destroy
+    session[:user_id] = nil
+    redirect_to root_path
+  end
+
+  private
+  def retrieve_oauth_user
```
Comment (sdevani) [inline app/controllers/sessions_controller.rb:17]: Test this to ensure it returns a user or nil. - add to sessions_spec.

What happens to `create` if `retrieve_oauth_user` returns nil

File: app/assets/stylesheets/sessions.css.scss [hunk 11]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Sessions controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
Comment (hainessss) [inline app/assets/stylesheets/sessions.css.scss]: Delete unused files

File: config/routes.rb [hunk 12]
```diff
@@ -1,4 +1,14 @@
 Crm::Application.routes.draw do
+
+  get 'auth/:provider/callback', to: 'sessions#create'
```
Comment (hainessss) [inline config/routes.rb:4]: Comment about what these routes are. They're all related to sign in/sign out

File: spec/helpers/sessions_helper_spec.rb [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+# describe SessionsHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
+describe SessionsHelper do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (hainessss) [inline spec/helpers/sessions_helper_spec.rb]: Remove this file

File: spec/models/user_spec.rb [hunk 14]
```diff
@@ -1,5 +1,40 @@
...
+    context "when a user doesn't exit" do
+      it "returns a new User object with info filled in" do
+        # shouldn't have to manually clear this, but not deleting the values created above^
+        User.destroy_all
+
+        auth_string = File.read('spec/models/response.rb')
+        auth = eval(auth_string)
+        auth_mash = Hashie::Mash.new(auth)
+
+        binding.pry
```
Comment (hainessss) [inline spec/models/user_spec.rb]: Get rid of binding.pry

**Pull request closed without merging**